examples/siro_sandbox/gui_navigation_helper.py

```diff
@@ -100,11 +100,86 @@ def viz_and_get_humanoid_walk_dir(self):
             path_points, path_endpoint_radius, path_color
         )
 
-        if len(path.points) >= 2:
-            walk_dir = mn.Vector3(path.points[1]) - mn.Vector3(path.points[0])
-            return walk_dir
+    def _get_humanoid_walk_dir_from_remote_gui_input(
+        self, visualize_path=True
+    ):
+        walk_dir = None
+        distance_multiplier = 0.0
+
+        (
+            target_pos,
+            target_rot_quat,
+        ) = self._sandbox_service.remote_gui_input.get_head_pose()
+
+        walk_dir, distance_multiplier = self._get_humanoid_walk_dir(
+            target_pos=target_pos,
+            target_rot_quat=target_rot_quat,
+            visualize_path=visualize_path,
+        )
+
+        return walk_dir, distance_multiplier
+
+    def _get_humanoid_walk_dir_from_ray_cast(self, visualize_path=True):
+        walk_dir = None
+        distance_multiplier = 0.0
+
+        target_on_floor = self._get_target_pos_from_ray_cast()
+        if target_on_floor is None:
+            return walk_dir, distance_multiplier
+
+        walk_dir, distance_multiplier = self._get_humanoid_walk_dir(
+            target_pos=target_on_floor,
+            target_rot_quat=None,  # habitat_sim.utils.common.random_quaternion() can be used to generate random rotations for testing
+            visualize_path=visualize_path,
+        )
+
+        return walk_dir, distance_multiplier
+
+    def _get_humanoid_walk_dir(
+        self, target_pos, target_rot_quat, visualize_path=True
+    ):
+        walk_dir = None
+        distance_multiplier = 0.0
+        geodesic_dist_threshold = 0.05
+
+        found_path, path = self._get_humanoid_walk_path_to(target_pos)
+        if (
+            found_path
+            and len(path.points) >= 2
+            and path.geodesic_distance >= geodesic_dist_threshold
```

Comment:
`HumanoidRearrangeController` doesn't move the agent exactly `|walk_dir|` in the `walk_dir` direction. It either overshoots or undershoots and we usually end up in the situations when agent is moving forward and backward in the proximity if the target position. Similar happens to rotations: if agent rotates on the spot it can move from side to side. That's why I use `geodesic_dist_threshold`.